### ISSUE TITLE
Deal with <ft-content title=""></ft-content>

### ISSUE THREAD

**Issue by i-like-robots** [Financial-Times/next-es-interface#338]
There are [many instances](http://next-elastic.ft.com/v3_api_v2/item/_search?q=bodyXML:%22%3Cft-content%22) of `<ft-content>` elements without a body. We have no code currently to handle these. Are they meant to be links?

**Comment by andygnewman** [comment]
I've had a look at a few of these on 'classic' ft.com and there seems to be a theme here.

They are images that act as links, ie. a structure of

```
<a>
    <img>
</a>
```

There seem to be examples where the link target is either the image itself at a larger size ("click to enlarge") or to a specific piece of content / article.

However in the XML that is coming from CAPI V2 the only reference we get (url) is specifically to the target, as it is in other `<ft-content>` elements.

What is missing is the content of the link - the image. It looks like this is not being handled by CAPI, so we should raise that as a card with them, then we can properly resolve within Next.

**Comment by andygnewman** [comment]
Some examples;

Image links that are click to enlarge; 
Classic: http://www.ft.com/cms/s/3/7acf12ba-9521-11e5-8389-7c9ccf83dceb.html
Next: https://next.ft.com/content/7acf12ba-9521-11e5-8389-7c9ccf83dceb

Image links that link to content;
Classic: http://www.ft.com/cms/s/0/44f0a19e-14ad-11e5-9509-00144feabdc0.html
Next: https://next.ft.com/content/44f0a19e-14ad-11e5-9509-00144feabdc0

**Comment by dgem** [comment]
Looking at the original XML in [methode](http://methode-api-01-pr-uk-p.svc.ft.com:8080/eom-file/7acf12ba-9521-11e5-8389-7c9ccf83dceb) (once decoded from base64) I see this pattern:
```
<p><a title="im.ft-static.com" href="http://im.ft-static.com/content/images/70f7caf0-9785-11e5-9228-87e603d47bdc.img"><web-inline-picture alt="" dtxInsert="Inline image (original low res)" fileref="/FT/Graphics/Online/Z_Undefined/2015/11/PPP%20success%20600.png?uuid=e7c27260-9786-11e5-9228-87e603d47bdc" tmx="600 526 600 526" align="center"/>
```
This appears in the contentv2 api as:
```
<ft-content title=\"im.ft-static.com\" type=\"http://www.ft.com/ontology/content/Article\" url=\"http://api.ft.com/content/70f7caf0-9785-11e5-9228-87e603d47bdc\">\n</ft-content>
```
So it would seem that CAPI v2 is replacing the anchor link with a link to the referenced content, however it also looks like that in doing so the inline image is being removed.

What is yet to be investigated is what happens with an inline image that is not inside an anchor tag. I would hope this case has been covered and it is processed appropriately.

It's standard bodyXML to convert links into replace the `<a>` with `<ft-content>`, but ordinarily the text of the link is preserved, however in these cases there is no text and the `<web-inline-picture>` is lost.

A bit more investigation into the `Methode Article Transformer` is required to isolate the bug...

In the meantime, please can @andygnewman, @matthew-andrews or @i-like-robots confirm whether this is coming from CAPIv1 or CAPIv2 ?

**Comment by dgem** [comment]
So although there are test cases for using inline images and others for anchor tags, there isn't a feature for inline images inside anchor tags. As such it seems this is not a tested scenario. Will discuss with dynamic publishing team as it seems like a genuine 'not tested' scenario, although one which it seems editorial is expected CAPv2 to support.

**Comment by matthew-andrews** [comment]
we only ever used rich body info from capi v2 so this will be v2 @dgem

**Comment by davidlintonattheft** [comment]
Our body XML format doesn't allow links inside links, where links in our conception is expanded to include embedded content.

As far as I can see we've isolated 2 occasions where an image is used as a link:
1. click for big
2. image from an article used to link to the article.

For 1., the link is to a MediaResource or an ImageSet. (UPP doesn't correctly identify these, but should.) The embedded image is just a small representation of the image -- you could implement it in Next by scaling down the linked image, Webapp already do this for large images.

For 2., it's more of a "related content" situation, where there be should be a visible link to the article in addition, otherwise the user won't have a good idea where they are going. This is certainly the case for the FirstFT example above. 

Mus has agreed that Editorial shouldn't be doing either of these things in future. But in neither case is the image _within _the link a vital part of the article.

For existing content, Max has suggested this:
a. If the linked content is an image or mediaresource and there is otherwise nothing within the <ft-content>, then the image should be displayed embedded (and Next should deal with any click-for-big functionality desired)
b. If the linked content is an article and there is otherwise nothing within <ft-content>, then the whole thing could be ignored, or, perhaps Next could go fetch the master image from the linked article -- this situation is similar to <ft-related> except that with <ft-content> there is more of an expectation that the link should be displayed _exactly here_ rather than off to one side. (I think UPP should be clearer about the distinction between <ft-related> and <ft-content>.)

I'll have a hunt through the search results to see if there are any other use cases for this kind of construction.

**Comment by davidlintonattheft** [comment]
Also by the way your search isn't looking for "<ft-content" only, there are many articles in there which just contain the phrase "FT content".

I get 333 results for[FT content type](http://next-elastic.ft.com/v3_api_v2/item/_search?q=bodyXML:%22%3Cft-content%20type%22) which is probably more accurate.

**Comment by andygnewman** [comment]
@davidlintonattheft Thanks.

So if I understand this correctly;

Scenario 1 (click for big) we would expect CAPI V2 to provide an `<ft-content>` tag purely for the image, rather than the empty tag as currently. The consumer of the CAPI v2 can then decide whether to implement a click for big.

Scenario 2 (image article link) we expect CAPI to provide nothing and will fully strip out the empty `<ft-content>` 

Are these changes planned in CAPI?

**Comment by i-like-robots** [comment]
Was there any resolution to this?

**Comment by davidlintonattheft** [comment]
It doesn't look like anything got done, sorry, and I missed Andy's follow-up.

The way things stand the link always points to the referenced thing (either the article, or the big image.) Max was suggesting what Next could do with this information, not adding new requirements for CAPI.

The change to CAPIv2 that does need to be done is to identify the linked content as an ImageSet/MediaResource, rather than an Article as currently. Right now you would need to use your own knowledge to identify the type of an item, as the <ft-content> element always reports "article" even when it isn't. This is [`UPP-1246`](https://jira.ft.com/browse/UPP-1246)

**Comment by i-like-robots** [comment]
Fixed/avoided by #626
